### ISSUE TITLE
In STS, sleuth causes the compile error: Build path is incomplete. Cannot find class file for org/tmatesoft/svn/core/wc2/SvnOperationFactory #667

### ISSUE THREAD

**Issue by martinlippert** [spring-projects/spring-ide#105]
(migrated from https://issuetracker.springsource.com/browse/STS-4437)

If you create a spring boot application and you add spring-cloud-config-server and spring-cloud-starter-sleuth dependencies, STS (after cleaned and rebuilt) produce the error:
"Build path is incomplete. Cannot find class file for org/tmatesoft/svn/core/wc2/SvnOperationFactory"
I have created the next simple project as an example:
https://github.com/cesargithub5/ConfigServer
Also, this issue was raised first in:
https://github.com/spring-cloud/spring-cloud-config/issues/667

(some additional comments and follow-ups can be found in the tracker story or https://issuetracker.springsource.com/browse/STS-4437)

**Comment by cabbonizio** [comment]
Hi @martinlippert this is a slightly different flavor of the error but very similar with Sleuth being involved:

> Class dependency error 'com/atomikos/jdbc/AtomikosDataSourceBean' occurred on aspect definition 'Aspect definition [\cart-retrieval-service\src\main\java\com\qvc\dcp\integration\services\cart\Application.java:-1] advise type [after] advise [org.springframework.cloud.sleuth.instrument.scheduling.TraceSchedulingAspect.traceBackgroundThread(org.aspectj.lang.ProceedingJoinPoint)]' while processing bean 'xaDataSourceWrapper (-1) [org.springframework.boot.jta.atomikos.AtomikosXADataSourceWrapper]'. Check if builder classpath is complete

Do you think this issue could also be the cause of the error I see in Eclipse?  I am on Oxygen with most up to date STS installed.   It seems there are no runtime issues at all just some pesky Eclipse compilation errors.

**Comment by martinlippert** [comment]
I am not yet sure what exactly causes this issue, but I can reproduce it (which is great).

If you want to quickly get rid of the errors, you can go to the `Preferences -> Spring -> Validation`, select the tab `Project Builders` and deactivate the `AOP Reference Model Builder`.

**Comment by cabbonizio** [comment]
Thanks @martinlippert I did figure this out already and have disabled the validation which is perfectly fine at this point.  I'll hang around on this thread to see how things progress.  I'm glad it's reproducible

**Comment by martinlippert** [comment]
Glad to hear you found the workaround already. Will take a deeper look at the root cause of this for the next release.

**Comment by cabbonizio** [comment]
Bravo!

**Comment by martinlippert** [comment]
I analyzed this in more detail. The reason for this error to appear is:

- Spring IDE creates a model for spring beans behind the scenes
- one of those beans is a bean for a SVN-based environment repository
- the bean is defined to be created at runtime automatically for your convenience if you have a specific bean profile active. This is, of course, ignored by Spring IDE while creating the beans model.
- As a second step, the AOP builder tries to create a model for aspects and their pointcut matches (to visualize that for the developer).
- while doing that, the AOP builder takes every bean and analyzes that bean, more accurately the type of the bean
- while doing this, it can't really analyze the bean because it uses types of that missing third-party lib, which is not on the classpath of the project (for a very good reason).
- that throws an exception.

It makes a lot of sense for the project and the Spring Cloud library to be defined that way. Therefore, we need to be more relaxed when analyzing beans and take into account that types might be missing. I added that to the AOP model builder. That also prevents this error marker from being created and attached to that project.
